api/src/main/java/org/pmiops/workbench/api/ConceptsController.java

```diff
@@ -46,16 +53,24 @@
             .vocabularyId(concept.getVocabularyId())
             .conceptSynonyms(concept.getSynonyms().stream().map(ConceptSynonym::getConceptSynonymName).collect(Collectors.toList()));
 
+  private static final Function<org.pmiops.workbench.cdr.model.VocabularyCount, VocabularyCount>
+      TO_CLIENT_VOCAB_COUNT =
+      (vocabCount) -> new VocabularyCount()
+          .conceptCount(vocabCount.getConceptCount())
+          .vocabularyId(vocabCount.getVocabularyId());
```

Comment:
This isn't intended to be WIP; I think this code is ready to go.

I could add domain ID, but it's already implied by the context in which this gets used. What's the point in returning the same domain ID for all the results?

Similarly, in the context in which this gets used, we either want standard counts or all concept counts... not both. Calculating both when we only want one hurts performance, since the columns you have to reference differ.

Reply:
Imagine the quick search page or concept search page where you are searching across all domains and showing cards with the concept results in each domain : 

You can fetch all of the results for each domain vocabulary counts in one query and build your domain info objects out of that for each domain.  So I guess i am talking more about a  backend api model change than a Client Model change so that DomainInfo can have a Vocabulary List relationship.

Reply:
So the plan for this PR is to have a List<DomainVocabularyInfo> object on each DomainInfo. Those will get used to populate these counts (choosing standard vs. all concept count based on the request.) DomainVocabularyInfo, as a child, does have a foreign key to the parent DomainInfo's primary key. But that doesn't need to be reflected in the API.

At the moment we're only fetching vocabulary counts for a single domain at a time -- the one in the request. (Which is good from a performance standpoint when there's keywords involved; we don't have to wait for the vocabulary counts for all the domains we're not looking at.) 

If we wanted to change that so that we fetched vocabulary info for all domains, I'd change it so that the DomainInfo object returned in the API response had the list of vocab counts under it. Is that what you're proposing?